ksqldb-streams/src/test/java/io/confluent/ksql/execution/streams/materialization/ks/KsMaterializedTableTest.java

```diff
@@ -122,56 +135,125 @@ public void shouldThrowIfGettingStateStoreFails() {
+    verify(kafkaStreams).query(queryTypeCaptor.capture());
+    StateQueryRequest request = queryTypeCaptor.getValue();
+    assertThat(request.getQuery(), instanceOf(RangeQuery.class));
+    RangeQuery rangeQuery = (RangeQuery)request.getQuery();
+    assertThat(rangeQuery.getLowerBound(), is(Optional.of(A_KEY)));
+    assertThat(rangeQuery.getUpperBound(), is(Optional.empty()));
   }
 
   @Test
-  public void shouldGetWithCorrectParams() {
+  public void shouldRangeQueryWithCorrectParams_upperBound() {
+    // Given:
+    when(kafkaStreams.query(any())).thenReturn(queryResult);
+    when(queryResult.getOnlyPartitionResult()).thenReturn(rangeResult);
+    when(rangeResult.getResult()).thenReturn(keyValueIterator);
+
     // When:
-    table.get(A_KEY, PARTITION);
+    table.get(PARTITION, null, A_KEY2);
 
     // Then:
-    verify(tableStore).get(A_KEY);
+    verify(kafkaStreams).query(queryTypeCaptor.capture());
+    StateQueryRequest request = queryTypeCaptor.getValue();
+    assertThat(request.getQuery(), instanceOf(RangeQuery.class));
+    RangeQuery rangeQuery = (RangeQuery)request.getQuery();
+    assertThat(rangeQuery.getLowerBound(), is(Optional.empty()));
+    assertThat(rangeQuery.getUpperBound(), is(Optional.of(A_KEY2)));
+  }
+
+  @Test
+  public void shouldRangeQueryWithCorrectParams_bothBounds() {
+    // Given:
+    when(kafkaStreams.query(any())).thenReturn(queryResult);
+    when(queryResult.getOnlyPartitionResult()).thenReturn(rangeResult);
+    when(rangeResult.getResult()).thenReturn(keyValueIterator);
+
+    // When:
+    table.get(PARTITION, A_KEY, A_KEY2);
+
+    // Then:
+    verify(kafkaStreams).query(queryTypeCaptor.capture());
+    StateQueryRequest request = queryTypeCaptor.getValue();
+    assertThat(request.getQuery(), instanceOf(RangeQuery.class));
+    RangeQuery rangeQuery = (RangeQuery)request.getQuery();
+    assertThat(rangeQuery.getLowerBound(), is(Optional.of(A_KEY)));
+    assertThat(rangeQuery.getUpperBound(), is(Optional.of(A_KEY2)));
+  }
+
+  @Test
+  public void shouldRangeQueryWithCorrectParams_noBounds() {
+    // Given:
+    when(kafkaStreams.query(any())).thenReturn(queryResult);
+    when(queryResult.getOnlyPartitionResult()).thenReturn(rangeResult);
+    when(rangeResult.getResult()).thenReturn(keyValueIterator);
+
+    // When:
+    table.get(PARTITION, null, null);
+
+    // Then:
+    verify(kafkaStreams).query(queryTypeCaptor.capture());
+    StateQueryRequest request = queryTypeCaptor.getValue();
+    assertThat(request.getQuery(), instanceOf(RangeQuery.class));
+    RangeQuery rangeQuery = (RangeQuery)request.getQuery();
+    assertThat(rangeQuery.getLowerBound(), is(Optional.empty()));
+    assertThat(rangeQuery.getUpperBound(), is(Optional.empty()));
   }
 
   @Test
   public void shouldReturnEmptyIfKeyNotPresent() {
+    // Given:
+    when(kafkaStreams.query(any())).thenReturn(queryResult);
+    when(queryResult.getOnlyPartitionResult()).thenReturn(keyResult);
```

Comment:
Similar to the above, we have no guarantee that this mocked behavior accurately reflects how a real StateQueryResult works when there's no key found. Returning a real `makeResult(null)` would be more airtight.